src/net.cpp

```diff
@@ -1080,7 +1080,12 @@ void ThreadSocketHandler()
                 }
             }
         }
-        if(vNodes.size() != nPrevNodeCount) {
+        size_t vNodesSize;
+        {
+            LOCK(cs_vNodes);
+            vNodesSize = vNodes.size();
+        }
+        if(vNodesSize != nPrevNodeCount) {
             nPrevNodeCount = vNodes.size();
```

Comment:
OK well this means that Bitcoin has implemented this very same code, thinking they have fixed unlocked access to vnodes.size, but in fact, they haven't.

Reference here - https://github.com/bitcoin/bitcoin/commit/dfed983f19e7b61da243362f00fa3a14c92dae45